### ISSUE TITLE
TypeScript: Provide setting to organize imports on format

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/TypeScript#46217]
Testing microsoft/vscode#48150

This would be awesome.

**Comment by andrewbranch** [comment]
@mjbvz not quite sure what the request is here. I guess the OP wants TypeScript’s “Format Document” command to be able to include organizing imports, but most of the follow-up comments are about organize imports clashing with prettier, which would be completely unaffected by any changes we make to formatting with TS Server.

**Comment by mjbvz** [comment]
Yes this is for the original request: an option for the `format` command on TS Server to automatically organize imports

This wouldn't fix any formatting conflicts with prettier but could make them a bit less painful since it would let format and organize imports happen in a single step instead of in two separate stages

**Comment by andrewbranch** [comment]
I think I’m still missing something with regard to how this would help with prettier. I’m assuming that people who use both prettier and TS’s organize imports have a config like:

```json
{
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
        "source.organizeImports": true
    }
}
```

So on save, a format command would be dispatched to prettier and an organize imports command would be dispatched to TS Server. TS Server would never see a format request in this scenario.
